src/github.rs

```diff
@@ -182,31 +117,94 @@ let token_res = &data.oauth
                 .json(e.to_string());
         }
         }
-        /*match userdetails(access_token.clone()).await {
-            Ok(val) => {    
-                //dbg!(val);
-                dbg!(val.get("login").clone());
-
-            },
-            Err(e) => {
-                dbg!(e);
-            },
-        };*/
-        //access_token.secret().to;
-
     }else{
-        /*let html = format!(
-            r#"<html>
-            <head><title>OAuth2 Test</title></head>
-            <body>
-                Gitlab user info:
-                <pre></pre>
-                <a href="/">Home</a>
-            </body>
-        </html>"#,);
-        actix_web::HttpResponse::Ok().body(html)*/
         return actix_web::HttpResponse::BadRequest().content_type("application/json")
         .json("No user details found.");
     }
 
-}
\ No newline at end of file
+}
+async fn userdetails(access_token:String) -> Result<serde_json::value::Value,awc::error::HttpError> {
+
+    let token_val=format!("{}{}", String::from("token "), access_token);
+    
+    let clientnew = awc::Client::new();
+    
+    let request = clientnew
+         .get("https://api.github.com/user")    // <- Create request builder
+         .insert_header(("User-Agent", "Actix-web"))
+         .insert_header(("accept", "application/json"))
+         .insert_header(("authorization", token_val.clone()));
+    
+    
+         match request.send().await{
+            Ok(mut val) => {
+    
+                match val.body().await{
```

Comment:
Instead of `awc`, let's use `reqwest` crate. We are already using the later in fpm and I do not want to add awc as a dependency.

Also you can directly call `val.json().await` instead of getting `.body()` and then explicitly using serde.